data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/sink/KafkaSink.java

```diff
@@ -97,10 +105,12 @@ private void doInitializeInternal() {
 
     @Override
     public void doOutput(Collection<Record<Event>> records) {
+        reentrantLock.lock();
```

Comment:
What is this lock protecting? Is it possible to use `synchronized(<object>) {` instead

Reply:
we have imlemented this as other sinks like s3 sink.